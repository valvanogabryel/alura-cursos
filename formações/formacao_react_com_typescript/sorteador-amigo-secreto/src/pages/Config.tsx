import Footer from "../components/Footer";
import Form from "../components/Form";
import ParticipantsList from "../components/ParticipantsList";

const Config = () => {
  return (
    <>
      <Form />
      <ParticipantsList />
      <Footer />
    </>
  );
}

export default Config;