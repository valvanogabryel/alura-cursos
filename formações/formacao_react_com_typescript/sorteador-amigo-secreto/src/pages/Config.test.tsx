import { render } from "@testing-library/react"
import React from "react";
import { RecoilRoot } from "recoil"
import Config from "./Config";

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockNavigate
  }
});

describe('a página de configuração', () => {
  test('deve ser renderizada corretamente', () => {
    const { container } = render(
      <RecoilRoot>
        <Config />
      </RecoilRoot>
    )

    expect(container).toMatchSnapshot();
  });
});