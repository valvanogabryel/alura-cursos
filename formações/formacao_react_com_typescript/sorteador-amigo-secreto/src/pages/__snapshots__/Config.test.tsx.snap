// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a página de configuração deve ser renderizada corretamente 1`] = `
<div>
  <form>
    <input
      placeholder="Insira os nomes dos participantes"
      type="text"
      value=""
    />
    <button
      disabled=""
      type="submit"
    >
      Adicionar
    </button>
    
  </form>
  <ul />
  <footer>
    <button
      disabled=""
    >
      Iniciar brincadeira
    </button>
  </footer>
</div>
`;
